### PR TITLE
add lobby player list

### LINKED ISSUE #53: Lobby Player List
Sprite 509, 2, 4, 5, Font OCR Extended

TODO:
- Create prefab
- Dynamically generate scrollable player list
- Make it possible to challenge other players (mock it up for now, on click show challenged sprite)
- Set up necessary signals and interfaces

### PR DESCRIPTION
**Pull request by Fruktus**
Closes #53

### CODE REVIEW

File: QuadradiusR/scripts/player_list.gd [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+func add_player(username: String, is_current = false):
+	var new_player = player_template.instance()
+	new_player.username = username
+
+	new_player.connect("player_hover_start", self, "_on_hover_start")
+	new_player.connect("player_hover_end", self, "_on_hover_start")
+	
+	if is_current:
+		new_player.is_self_player = true
```
Comment (Observer46) [inline QuadradiusR/scripts/player_list.gd:30]: If we decide use constructors then this is not needed to set `is_self_player` - should be done in constructor.

File: QuadradiusR/scripts/player_list.gd [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+func remove_player(username: String):
+	for child in player_vbox.get_children():
+		if child.username == username:
+			player_left_sfx.play()
+			child.queue_free()
+
+
+func _on_hover_start(username):
```
Comment (Observer46) [inline QuadradiusR/scripts/player_list.gd:44]: `username: String`

File: QuadradiusR/scripts/player_list.gd [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	emit_signal("player_hover_start", username)
+
+
+func _on_hover_end(username):
```
Comment (Observer46) [inline QuadradiusR/scripts/player_list.gd:48]: `username: String`

File: QuadradiusR/scripts/player_list_row.gd [hunk 4]
```diff
@@ -0,0 +1,55 @@
+extends PanelContainer
+
+signal player_hover_start(username)
+signal player_hover_end(username)
+signal player_clicked(username)
+
+export var username = "asd GUEST"
```
Comment (Observer46) [inline QuadradiusR/scripts/player_list_row.gd:7]: I think it is better to move all `export var` variables closer to `var` variables as they might server as scene attributes set in constructor. Probably better to have `export var` above standard var`.

**Review by Observer46 — CHANGES_REQUESTED** [review]
Overall it is very nice, but some I think some changes needs to be done related to the constructors.

File: QuadradiusR/scripts/player_list.gd [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+		if child.username == username:
+			child.receive_challenge()
+
+
+func add_player(username: String, is_current = false):
+	var new_player = player_template.instance()
+	new_player.username = username
+
+	new_player.connect("player_hover_start", self, "_on_hover_start")
+	new_player.connect("player_hover_end", self, "_on_hover_start")
```
[the remark this answers is not in the document]
Comment (Fruktus) [reply QuadradiusR/scripts/player_list.gd:27]: XD Good catch

File: QuadradiusR/scripts/player_list.gd [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+
+func receive_challenge(username):
+	for child in player_vbox.get_children():
+		if child.username == username:
+			child.receive_challenge()
+
+
+func add_player(username: String, is_current = false):
+	var new_player = player_template.instance()
```
[the remark this answers is not in the document]
Comment (Fruktus) [reply QuadradiusR/scripts/player_list.gd:23]: very nice idea, will lead to better codebase

**Review by Observer46 — APPROVED** [review]
It's okey mate!

**Pull request merged**